sentence_transformers/sampler.py

```diff
@@ -53,32 +72,33 @@ def __init__(
         if self.batch_size % 2 == 1:
             raise ValueError("The batch size for `GroupByLabelBatchSampler` must be divisible by 2.")
 
-        for column_name in valid_label_columns or []:
-            if column_name in dataset.column_names:
-                labels = dataset["label"]
-                break
-        else:
-            raise ValueError(f"None of the valid_label_columns {valid_label_columns} are in the dataset.")
-
-        del dataset
+        labels = self._determine_labels_to_use(dataset, valid_label_columns)
         groups = defaultdict(list)
         for sample_idx, label in enumerate(labels):
             groups[label].append(sample_idx)
 
         self.groups = {
             label: sample_indices[:num_samples]
             for label, sample_indices in groups.items()
-            if (num_samples := len(sample_indices) // 2)
+            if (num_samples := len(sample_indices) // 2 * 2)
         }
 
+    @staticmethod
+    def _determine_labels_to_use(dataset: "Dataset", valid_label_columns: List[str]) -> List[Any]:
+        for column_name in valid_label_columns or []:
+            if column_name in dataset.column_names:
+                return dataset[column_name]
+        else:
+            raise ValueError(f"None of the valid_label_columns {valid_label_columns} are in the dataset.")
```

Comment:
Now that you've moved this into a separate method, I think we can simplify this a bit further by removing the `else`. I've also expanded the error to include the column names, as that can be helpful.
```suggestion
        raise ValueError(
            f"None of the valid_label_columns {valid_label_columns} are in the dataset, "
            f"which only has these columns: {dataset.column_names}."
        )
```